pyiceberg/table/snapshots.py

```diff
@@ -116,3 +144,199 @@ class MetadataLogEntry(IcebergBaseModel):
+        def set_non_zero(properties: Dict[str, str], num: int, property_name: str) -> None:
+            if num > 0:
+                properties[property_name] = str(num)
+
+        properties: Dict[str, str] = {}
+        set_non_zero(properties, self.added_size, 'added-files-size')
+        set_non_zero(properties, self.removed_size, 'removed-files-size')
+        set_non_zero(properties, self.added_files, 'added-data-files')
+        set_non_zero(properties, self.removed_files, 'removed-data-files')
+        set_non_zero(properties, self.added_eq_delete_files, 'added-equality-delete-files')
+        set_non_zero(properties, self.removed_eq_delete_files, 'removed-equality-delete-files')
+        set_non_zero(properties, self.added_pos_delete_files, 'added-position-delete-files')
+        set_non_zero(properties, self.removed_pos_delete_files, 'removed-position-delete-files')
+        set_non_zero(properties, self.added_delete_files, 'added-delete-files')
+        set_non_zero(properties, self.removed_delete_files, 'removed-delete-files')
+        set_non_zero(properties, self.added_records, 'added-records')
+        set_non_zero(properties, self.deleted_records, 'deleted-records')
+        set_non_zero(properties, self.added_pos_deletes, 'added-position-deletes')
+        set_non_zero(properties, self.removed_pos_deletes, 'removed-position-deletes')
+        set_non_zero(properties, self.added_eq_deletes, 'added-equality-deletes')
+        set_non_zero(properties, self.removed_eq_deletes, 'removed-equality-deletes')
+
+        return properties
+
+
+properties = ['records', 'files-size', 'data-files', 'delete-files', 'position-deletes', 'equality-deletes']
+
+
+def truncate_table_summary(summary: Summary, previous_summary: Mapping[str, str]) -> Summary:
+    for prop in {
+        'total-data-files',
+        'total-delete-files',
+        'total-records',
+        'total-files-size',
+        'total-position-deletes',
+        'total-equality-deletes',
+    }:
+        summary[prop] = '0'
+
+    if value := previous_summary.get('total-data-files'):
+        summary['deleted-data-files'] = value
+    if value := previous_summary.get('total-delete-files'):
+        summary['removed-delete-files'] = value
+    if value := previous_summary.get('total-records'):
+        summary['deleted-records'] = value
+    if value := previous_summary.get('total-files-size'):
+        summary['removed-files-size'] = value
+    if value := previous_summary.get('total-position-deletes'):
+        summary['removed-position-deletes'] = value
+    if value := previous_summary.get('total-equality-deletes'):
+        summary['removed-equality-deletes'] = value
+
+    return summary
+
+
+def merge_snapshot_summaries(
+    summary: Summary,
+    previous_summary: Optional[Mapping[str, str]] = None,
+) -> Summary:
+    if summary.operation == Operation.OVERWRITE and previous_summary is not None:
+        summary = truncate_table_summary(summary, previous_summary)
```

Comment:
Just to confirm my understanding: Combining with the #41, It seems in python we treat `Operation.OVERWRITE` as overwriting all the data files in the previous snapshot.

If my understand is correct, I have a follow-up question: Do we plan to support overwritting a single data file like
```java
OverwriteFiles overwrite = table.newOverwrite();
overwrite.deleteFiles(oldDataFile);
overwrite.addFile(newDataFile);
overwrite.commit();
```
in the future?

Reply:
@HonahX Yes, I think we want to add this in the future. I kept the current version as minimal as possible. In #41 we're writing a single file right now. So that would always replace all the other files when doing an overwrite.

Once we go to writing to partitioned tables, we'll need to implement this when overwriting a specific partition.

Reply:
@Fokko, while I understand the logic of trying to keep this simple, it worries me that this code assumes that overwrite completely replaces the table. If the operation does actually truncate a table, then the caller should call a `truncate` method to inform the summary code that it should do this, just like it calls `add_file` and `remove_file`. Triggering that behavior with `overwrite` is just too broad.

Reply:
With Spark it is not common to overwrite a full table, but I would expect this with PyIceberg to be more common. Typically you load the whole dataset into memory, wrangle it, and write it back (or at least store a copy of it, to know on which data a model was trained).

I've added the flag 👍